docs/design/operator.md

```diff
@@ -0,0 +1,200 @@
+
+#### Volume Density
+
+The Gluster operator will monitor the total volume density of any cluster it is
+automating. It will trigger events when the number of volumes in the cluster
+reaches a pre-defined level that should prompt corrective action. This value
+will be configurable, and should be set to a value that will allow for
+corrective action to take place before cluster nodes lag or crash.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Device Health
+
+The Gluster operator will monitor the health of all devices in an automated
+Gluster cluster. It will report a "Warning" when a device reports failing health
+and will report a "Critical" problem when a device becomes inaccessible. Support
+for this feature may vary form one device type to another.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Volume Health
+
+The Gluster operator will monitor the health of all volumes in an automated
+Gluster cluster. It will report a "Warning" when a volume reports failing bricks
+and will report a "Critical" problem when a volume becomes inaccessible.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Node Health
+
+The Gluster operator will monitor the health of all nodes in an automated
+Gluster cluster. It will report a "Critical" problem when a node becomes
+inaccessible.
+
+**Target Configurations:** External, Hosted, Managed
+
+### Cluster Management
+
+In response to the monitoring events above, the Gluster operator will take any
+number of corrective management actions.
+
+#### Storage Scaling
+
+In the event that the overall cluster reports a critical capacity level, the
+operator will deploy a set number of additional storage nodes with new storage
+devices attached. The number of additional nodes to be deployed will be
+configurable.
+
+**Target Configurations:** Managed
+
+#### Cluster Scaling
+
+In the event that the overall cluster reports a critical volume density level,
+the operator will deploy a new cluster of Gluster nodes to receive volume
+creation requests. The number of additional nodes to be deployed will be
+configurable.
+
+**Target Configurations:** Managed
+
+### Disaster Recovery
+
+In response to the health monitoring events above, the Gluster operator will
+take any number of corrective actions to recover from a situation that could
+result in potential data loss.
+
+#### Brick Replacement
+
+If a volume reports a failing brick, the Gluster operator will try to create a
+new brick and use it as a replacement for the failing one.
+
+**Target Configurations:** External, Hosted, Managed
+
+#### Device Replacement
+
+If a node reports a failing device, the Gluster operator will try to provision a
+new device and attach it to the node to replace the failing device. It would
+then recreate any bricks on the failing device and trigger a brick replacement
+for any affected volumes.
+
+**Target Configurations:** Managed
+
+#### Node Replacement
```

Comment:
Here again, there is a split between the operator's tasks and Gluster's internal responsibilities. The operator would create a new node, and once data has been rebuilt, destroy the old node. Gluster would internally handle the brick reconstruction.